src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
+                // then we can be helpful and convert that to the directory rather than forcing
+                // the command to be re-typed.
+                string ilcDir = IlcPath;
+                if(File.Exists(IlcPath) && !Directory.Exists(IlcPath))
+                {
+                    string potentialIlcDir = Path.GetDirectoryName(IlcPath);
+                    if(Directory.Exists(potentialIlcDir))
+                    {
+                        ilcDir = potentialIlcDir;
+                    }
+                }
+                config.IlcPath = ilcDir;
+                config.IlcSdkPath = ilcDir;
             }
```

Comment:
I agree that the naming isn't ideal, though not planning to change that part. 

The second Directory.Exists is largely me being careful. I can't think of any specific case where it would fail, but there are an awful lot of obscure path syntax rules out there across all our platforms. Doing the check seems like a very small price to pay in terms of perf/complexity to make sure we aren't making things worse.

I'll add the spaces, thanks!